examples/database/cosmosdb/account.yaml

```
@@ -0,0 +1,34 @@
+---
+apiVersion: azure.crossplane.io/v1alpha3
+kind: ResourceGroup
+metadata:
+  name: crossplane-example
+spec:
+  location: West US 2
+  reclaimPolicy: Retain
+  providerRef:
+    name: azure-provider
+---
+apiVersion: database.azure.crossplane.io/v1beta1
+kind: CosmosDBAccount
+metadata:
+  name: mymongoaccount
+  annotations:
+    crossplane.io/external-name: mymongodbaccount
+spec:
+  forProvider:
+      resourceGroupName: crossplane-example
```

```suggestion
      resourceGroupNameRef:
            name: crossplane-example
```

This is functionally equivalent to what you already had, but will exercise the reference resolution code by using a reference to the above `kind: ResourceGroup` to figure resolve the (external) `resourceGroupName`.